tests/testsuite/main.rs

```diff
@@ -132,6 +132,8 @@ mod weak_dep_features;
 mod workspaces;
 mod yank;
 
+find_tests!("tests/snapshots/");
```

Comment:
Should we run the macro in `main` or in each test file?

Reply:
I  could see both being fine. It might fix the problems with naming though to have them in each test file. It could also allow for partial ports of some test files

Reply:
Being in `main` means its one-and-done, nothing to do when adding more tests

Being in `cargo_add` means
- We tests are listed to the user under `cargo_add`
- We don't have to be as free in discovering all tests anywhere but `import_tests` can be looking for a lot stricter of a pattern
- We can access `init_registry` without it being `pub(crate)` (I think)

If `re-run-if-changed` accepted globs, we could change our folder hierarchy:
```
tests/testsuite/cargo_add.rs -> tests/testssuite/cargo_add/mod.rs
tests/snapshots/add/* tests/testsuite/cargo_add/*
```
This would mean that we could just do `import_tests!(".");` or `import_tests!("*/mod.rs");`
- We wouldn't need the `path` attribute / things work more normally
- Things are even closer together and easier to compare among test areas

Reply:
In that case, would it make sense to support globs in `re-run-if-changed` first?

Reply:
I'd say they should be looked into in parallel.  We shouldn't block incremental improvement to our testing our on a user-facing change that might take some time to work out.

Reply:
That works for me, I'll move the call to `cargo_add`